ui/src/main/webapp/js/models/Service.js

```diff
@@ -133,35 +133,42 @@ define(['backbone', 'jquery','backboneassociations'],function (Backbone, $) {
             return deferred;
         },
         destroy: function() {
-            var deferred = $.Deferred(),
-                model = this;
-            var deleteUrl = [model.configUrl, "delete", model.get('properties').get("service.pid")].join("/");
+            var deleteUrl = [
+                this.configUrl,
+                "delete",
+                this.get('properties').get("service.pid")
+            ].join("/");
 
             return $.ajax({
                 type: 'GET',
                 url: deleteUrl
-            }).done(function (result) {
-                  deferred.resolve(result);
-                }).fail(function (error) {
-                    deferred.fail(error);
-                });
+            });
+        },
+        initializeFromModel: function (model) {
+            if (model.get("factory")) {
+                return this.initializeFromMSF(model);
+            } else {
+                return this.initializeFromService(model);
+            }
         },
         initializeFromMSF: function(msf) {
-            this.set({"fpid":msf.get("id")});
-            this.set({"name":msf.get("name")});
-            this.get('properties').set({"service.factoryPid": msf.get("id")});
-            this.initializeFromService(msf);
+            this.set({"fpid": msf.get("id")})
```

Comment:
Instead of calling set twice, call it once and pass in an object with both properties in it.  This will reduce the event change on model changes.